lasy/utils/mode_decomposition.py

```diff
@@ -0,0 +1,133 @@
+from lasy.profiles.transverse.transverse_profile import TransverseProfile
+from lasy.profiles.transverse.transverse_profile_from_data import (
+    TransverseProfileFromData,
+)
+from lasy.profiles.transverse.hermite_gaussian_profile import (
+    HermiteGaussianTransverseProfile,
+)
+from lasy.utils.exp_data_utils import find_d4sigma
+
+import numpy as np
+import math
+
+
+def hermite_gauss_decomposition(laserProfile, n_x_max=12, n_y_max=12):
+    """
+    Decomposes a `lasy` laser profile into a set of hermite-gaussian
+    modes.
+
+    The function takes either an instance of `TransverseProfile` or an
+    instance of `Laser` (that is, either a transverse profile or the
+    full 3D laser profile defined on a grid). In the case that an
+    instance of `Laser` is passed then the intensity of this profile
+    is projected onto an x-y plane for the decomposition.
+
+    Parameters
+    ----------
+    laserProfile: class instance
```

Comment:
I think this space is there in the numpy docstrings, right?
```suggestion
    laserProfile : class instance
```